filters/MetadataFilter.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace humhub\modules\legal\filters;
+
+use Yii;
+
+/**
+ * Class MetadataFilter
+ * Provides methods to filter out unneeded metadata from the provided content.
+ *
+ * @package humhub\modules\legal\filters
+ */
+class MetadataFilter
+{
+    /**
+     * Filter out unneeded metadata from the provided content.
+     *
+     * This method filters out unnecessary metadata from each content item, such as comments not belonging to the current user.
+     *
+     * @param array $content The array of content items to be filtered.
+     * @return array The filtered content array with unneeded metadata removed.
+     */
+    public static function filterMetadata($content)
```

Comment:
@ArchBlood Looks good.

I would consistently filter out all comments here.

Own comments are also exported via the comments anyway.

Reply:
> @ArchBlood Looks good.
> 
> I would consistently filter out all comments here.
> 
> Own comments are also exported via the comments anyway.

Once I'm back at the office I'll get to work on this.

Reply:
Done!

Reply:
Later today I'll also implement so that files are also excluded as we also use the File & Like APIs.

Reply:
I've now filtered out what URLs the API allows, currently user & file URL can't be filtered out, I believe this is because of how they're structured?